packages/@aws-cdk/aws-cognito/lib/user-pool.ts

```diff
@@ -1041,6 +1045,33 @@ export class UserPool extends UserPoolBase {
   }
 }
 
+interface UserPoolAttributes {
+  /**
+   * The physical ID of this user pool resource
+   */
+  userPoolId: string;
+  /**
+   * The ARN of the user pool
+   */
+  userPoolArn: string
+}
+
+class ImportedUserPool extends UserPoolBase {
```

Comment:
Could you try and retain the original structure of keeping the 'Import' class local to the `fromXXX()` methods?

You should be able to move the full implementation to within `fromUserPoolArn()` and call this method from `fromUserPoolId()`. Something like -


```ts
public static fromUserPoolId(...) {
  return fromUserPoolArn(..., Stack.of(scope).formatArn(...));
}
```

Reply:
@nija-at Are we sure we want to do this ?
I'm asking for the two following reasons:

- I've been asked in the past to extract the Import class (That was for EventBus) and it's done that way in a few other modules (search for `class Imported` in the codebase).
- I would need to find a different name for scope and id since I'm getting [eslint(@typescript-eslint/no-shadow)](https://github.com/typescript-eslint/typescript-eslint/blob/v4.16.1/packages/eslint-plugin/docs/rules/no-shadow.md) eslint error.

Reply:
I prefer the class to be inline since this acts as an added check to ensure that we don't export this class into the public API. If we extracted this out, it would be easy to just add an 'export' keyword and could easily be missed in reviews. However if you feel strongly on extracting this, that's fine with me. 

However, I would either rename of not use the `UserPoolAttributes` interface. This is reserved for specific types of import APIs - https://github.com/aws/aws-cdk/blob/master/docs/DESIGN_GUIDELINES.md#from-attributes - and I wouldn't want this to be incorrectly used.

Reply:
That makes sense. I'm moving the Imported class within the `fromUserPoolArn`. Thanks for the explanation @nija-at